### PR TITLE
Add libedit support

### LINKED ISSUE #38: libedit support

### PR DESCRIPTION
**Pull request by nbyavuz**
Fixes #38

### CODE REVIEW

File: meson.build [hunk 1]
```diff
@@ -659,55 +659,80 @@ endif
...
+      'readline_h': 'editline/readline.h',
+      'readline_h_flag': 'HAVE_EDITLINE_READLINE_H'
+    }
+    readline_flags = {
+      'history_h': 'readline/history.h',
+      'history_h_flag': 'HAVE_READLINE_HISTORY_H',
+      'readline_h': 'readline/readline.h',
+      'readline_h_flag': 'HAVE_READLINE_READLINE_H'
+    }
+    raw_flags = {
```
Comment (anarazel) [inline meson.build:697]: What does "raw" mean here?

Comment (nbyavuz) [reply meson.build:697]: I could not find alternative name. Maybe "default" or any suggestions?

Comment (anarazel) [reply meson.build:697]: I'm not even quite sure what you're trying to do - it's just the check for readline installed not inside readline/?

Comment (nbyavuz) [reply meson.build:697]: Let me try to explain.

There are two available options, user will prefer:

1. libedit over readline (if `libedit_preferred` is true)
2. readline over libedit (default)

In autoconf configuration, if libedit is preferred; checking for `history.h and readline.h` order is

1. editline
2. without editline and readline
3. readline


if readline is preferred, order is

1. readline
2. without editline and readline
3. editline

So, I created these variables namely `libedit_flags, readline_flags and raw_flags(which is for without editline and readline)`.
And I set the order of these variables based on `libedit_preferred` meson option for checking these `history.h and readline.h`.

Does that look good or do you prefer something else?

File: meson.build [hunk 2]
```diff
@@ -717,55 +717,80 @@ endif
...
+      endif
+
+      # Check readline.h
+      if not readline_h_found and cc.has_header(flag['readline_h'],
+          args: g_c_args, include_directories: g_c_inc,
+          dependencies: [readline], required: false)
+        readline_h = flag['readline_h']
+        cdata.set(flag['readline_h_flag'], 1)
+        readline_h_found = true
+      endif
```
Comment (anarazel) [inline meson.build:786]: This seems extremely verbose. This code block is near-identically duplicated, libedit_flags / readline_flags / raw_flags is near identical.

E.g. the header check above could be deduplicated by doing '@0@_h'.format(header). And I think something similar should be doable with libedit_flags, readline_flags, raw_flags.

Comment (nbyavuz) [reply meson.build]: I tried to implement the changes you recommended however, the previous version was more comprehensible in my opinion. I think the code became hard to read after the changes. What do you think?

**Pull request merged**